openassessment/fileupload/api.py

```diff
@@ -216,7 +218,38 @@ def append_uploads(self, *new_uploads):
         new_sizes = existing_file_sizes + sizes_to_add
         self._set_file_sizes(new_sizes)
 
-        return self._file_uploads_from_list_fields(new_descriptions, new_names, new_sizes)
+        new_file_uploads = self._file_uploads_from_list_fields(new_descriptions, new_names, new_sizes)
+
+        if self.block.is_team_assignment():
+            existing_file_upload_key_set = {
```

Comment:
Another way to do this would be `existing_file_upload_key_set = {upload.key for upload in self.get_uploads()}` and you just have to invoke it prior to line 213.

Reply:
...not that it really helps that much, but maybe makes it a tiny bit prettier...

Reply:
I don't think that will work. Since we've already updated the size/descriptions/names that would include the new files as well.